taqueria-vscode-extension/src/test/suite/extension.test.ts

```diff
@@ -5,102 +5,112 @@ import * as vscode from 'vscode';
+	'window.showQuickPick': () => Promise.resolve([choosePlugin]),
+	'window.showOpenDialog': () => {
+		return Promise.resolve([Uri.file(getNextOpenDialogUri())]);
+	},
+};
+vscodeMock = MockedObject.make(vscode, mockedMethods);
+
+describe('Extension Test Suite', async () => {
 	before(async () => {
 		const context: ExtensionContext = {
 			subscriptions: [],
 		} as any;
-		await taqueriaExtension.activate(context, {
-			vscode: MockedObject.make(vscode, {
-				'window.showInformationMessage': (msg: string) => Promise.resolve(console.log(msg)),
-			}),
-		});
-		await fse.copy(testProjectSource, testProjectDestination);
+
+		await taqueriaExtension.activate(context, { vscode: vscodeMock });
 
 		vscode.window.showInformationMessage('Start all tests.');
 	});
 
-	it('Verify that Taqueria Initiate will init new taquifed  project ', async () => {
+	it('Verify that Taqueria commands present in the command pallet list ', async () => {
 		await vscode.commands.getCommands(true).then(allCommands => {
 			const taqCommands = allCommands.filter(command => command.toLowerCase().includes('taq'));
 			assert.notEqual(taqCommands, undefined);
 		});
-
-		// TODO: Need to figure out if it is possible to interact with showDialogBox
-		//      If no this test cannot be automated
-		//      I tried to use RobotJS, but it has an issue to work with parser properly
-		//      Even if I exclude it it keeps asking to rebuild it due to an old version of npm
-		//      https://stackoverflow.com/questions/46384591/node-was-compiled-against-a-different-node-js-version-using-node-module-versio
-
-		// await vscode.commands.executeCommand("taqueria.init").then( ()=>{
-		// });
-
-		// Example how to use sleep to stop VS Code for a moment
-		// await sleep(60000);
 	});
 
-	// TODO: https://github.com/ecadlabs/taqueria/issues/605
-	it.skip('Verify that VS Code command Taqueria Scaffold will scaffold a project', async () => {
-		// Open taquified project in workspace under testProjectDestination
+	it('Verify that VS Code command Taqueria Compile Ligo will compile Ligo contract', async () => {
+		// It creates another process
+		// https://stackoverflow.com/questions/51385812/is-there-a-way-to-open-a-workspace-from-an-extension-in-vs-code
+		// await vscodeMock.commands.executeCommand('vscode.openFolder', vscode.Uri.parse(testProjectDestination));
+		// await workspace.updateWorkspaceFolders(0, 1, { uri: Uri.parse()});
 
-		// Run taqueria scaffold command
-		await vscode.commands.executeCommand('taqueria.scaffold');
-	});
+		setOpenDialogMocks(`${ligoContractFileDestination}`);
 
-	// TODO: https://github.com/ecadlabs/taqueria/issues/686
-	it.skip('Verify that VS Code command Taqueria Start Sandbox starts flextesa sandbox', async () => {
-		// Install plugin
-		await vscode.commands.executeCommand('taqueria.install @taqueria/plugin-flextesa');
+		await fse.rm(testProjectDestination, { recursive: true });
```

Comment:
Why are you removing the test project destination and then making it within the test? Shouldn't the project destination be made by the run-tests.sh script before the tests are run?

Reply:
Yes, but for each test, we do a clean-up of the folder - rm and mkdir on lines 84,85 are doing it for us. Because we want to have it clean for each test without plugins, etc. But we have to create a reference first (that is what we do in run-tests.sh, otherwise it won't be picked up by VS Code)